### PR TITLE
feat: add security

### LINKED ISSUE #4: Legg til spring-security

### PR DESCRIPTION
**Pull request by NilsOveTen**
resolves #4

### CODE REVIEW

File: src/main/kotlin/no/fdk/concept_catalog/security/SecurityConfig.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            ))
+        return jwtDecoder
+    }
+}
+
+private class TurtleMatcher: RequestMatcher{
+    override fun matches(request: HttpServletRequest?): Boolean =
+        request?.getHeader("Accept") != null &&
+            request.getHeader("Accept") == "text/turtle" &&
+            request.method == "GET"
```
Comment (azaxarov) [inline src/main/kotlin/no/fdk/concept_catalog/security/SecurityConfig.kt:58]:
```suggestion
        request?.getHeader(HttpHeaders.ACCEPT)?.contains(""text/turtle"") && request.method == HttpMethods.GET
```

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -4,3 +4,20 @@ logging:
   level.no: ${LOG_LEVEL:DEBUG}
   level.org.springframework: WARN
   level.org.springframework.web: WARN
+spring:
+  security.oauth2.resourceserver.jwt.jwk-set-uri: ${SSO_HOST}/auth/realms/fdk/protocol/openid-connect/certs
+security:
+  fdkIssuer: ${SSO_HOST:https://sso.staging.fellesdatakatalog.digdir.no}/auth/realms/fdk
```
Comment (azaxarov) [inline src/main/resources/application.yml:10]:
```suggestion
spring:
  security.oauth2.resourceserver.jwt.jwk-set-uri: ${SSO_HOST}/auth/realms/fdk/protocol/openid-connect/certs
  security.oauth2.resourceserver.jwt.issuer-uri: ${SSO_HOST:https://sso.staging.fellesdatakatalog.digdir.no}/auth/realms/fdk
```

File: src/main/kotlin/no/fdk/concept_catalog/security/SecurityProperties.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package no.fdk.concept_catalog.security
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+
+@ConstructorBinding
+@ConfigurationProperties("security")
+data class SecurityProperties(
```
Comment (azaxarov) [inline src/main/kotlin/no/fdk/concept_catalog/security/SecurityProperties.kt:8]: Denne data class er ikke nødvendig

File: src/main/kotlin/no/fdk/concept_catalog/security/SecurityConfig.kt [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @Bean
+    open fun jwtDecoder(properties: OAuth2ResourceServerProperties): JwtDecoder? {
+        val jwtDecoder = NimbusJwtDecoder.withJwkSetUri(properties.jwt.jwkSetUri).build()
+        jwtDecoder.setJwtValidator(
+            DelegatingOAuth2TokenValidator(
+                listOf(
+                    JwtTimestampValidator(),
+                    JwtIssuerValidator(securityProperties.fdkIssuer),
```
Comment (azaxarov) [inline src/main/kotlin/no/fdk/concept_catalog/security/SecurityConfig.kt:46]:
```suggestion
                    JwtIssuerValidator(roperties.jwt.issuerUri),
```

File: src/main/kotlin/no/fdk/concept_catalog/security/SecurityConfig.kt [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter
+import org.springframework.security.oauth2.core.DelegatingOAuth2TokenValidator
+import org.springframework.security.oauth2.jwt.*
+import org.springframework.security.web.util.matcher.RequestMatcher
+import javax.servlet.http.HttpServletRequest
+
+@Configuration
+open class SecurityConfig(
+    private val securityProperties: SecurityProperties
+) : WebSecurityConfigurerAdapter() {
```
Comment (azaxarov) [inline src/main/kotlin/no/fdk/concept_catalog/security/SecurityConfig.kt:17]:
```suggestion
open class SecurityConfig() : WebSecurityConfigurerAdapter() {
```

**Review by azaxarov — CHANGES_REQUESTED** [review]

File: src/main/resources/application.yml (hunk 2 above)
Comment (NilsOveTen) [reply src/main/resources/application.yml]: ce0a85d

File: src/main/kotlin/no/fdk/concept_catalog/security/SecurityProperties.kt (hunk 3 above)
Comment (NilsOveTen) [reply src/main/kotlin/no/fdk/concept_catalog/security/SecurityProperties.kt]: ce0a85d

File: src/main/kotlin/no/fdk/concept_catalog/security/SecurityConfig.kt (hunk 4 above)
Comment (NilsOveTen) [reply src/main/kotlin/no/fdk/concept_catalog/security/SecurityConfig.kt]: ce0a85d

File: src/main/kotlin/no/fdk/concept_catalog/security/SecurityConfig.kt (hunk 5 above)
Comment (NilsOveTen) [reply src/main/kotlin/no/fdk/concept_catalog/security/SecurityConfig.kt]: ce0a85d

File: src/main/kotlin/no/fdk/concept_catalog/security/SecurityConfig.kt (hunk 1 above)
Comment (NilsOveTen) [reply src/main/kotlin/no/fdk/concept_catalog/security/SecurityConfig.kt]: c73575f

**Review by azaxarov — APPROVED** [review]

**Pull request merged**